pkg/controller/storagecluster/reconcile.go

```diff
@@ -172,6 +176,29 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 		if phaseErr != nil {
 			reqLogger.Error(phaseErr, "Failed to set PhaseDeleting")
 		}
+
+		// Check if the storagecluster is labelled for data deletion, if yes
+		// label the child resources for data deletion too before issuing delete
+		if v, found := instance.ObjectMeta.Labels[CleanupPolicyLabel]; found {
+			if v == CleanupPolicyDelete {
+				cephCluster := &cephv1.CephCluster{}
+				err = r.client.Get(context.TODO(), types.NamespacedName{Name: generateNameForCephCluster(instance), Namespace: instance.Namespace}, cephCluster)
+				if err != nil {
+					if errors.IsNotFound(err) {
+						reqLogger.Info("CephCluster not found")
+					} else {
+						return reconcile.Result{}, fmt.Errorf("Unable to retrive cephCluster: %v", err)
+					}
+				} else {
+					cephCluster.Spec.CleanupPolicy.Confirmation = CleanupPolicyDelete
```

Comment:
```suggestion
					cephCluster.Spec.CleanupPolicy.Confirmation = cephv1.DeleteDataDirOnHostsConfirmation
```

Reply:
I have update the code to use the type when referring to rook spec. Done.